tests/unittests/test_salesforce_null_bookmark.py

```diff
@@ -0,0 +1,37 @@
+import unittest
+from unittest import mock
+import tap_salesforce
+from tap_salesforce import Salesforce, metrics
+from tap_salesforce.sync import sync_records
+import json
+
+
+
+class TestNullBookmarkTesting(unittest.TestCase):
+    @mock.patch('tap_salesforce.salesforce.Salesforce.query', side_effect=lambda test1, test2: [])
+    def test_not_null_bookmark_for_incremental_stream(self, mocked_query):
+        """
+        To ensure that after resolving write bookmark logic not get "Null" as replication key in state file, 
+        When we have selected incremental stream as Full table stream
+
+        """
+        sf = Salesforce(
+            refresh_token="test",
+            sf_client_id="test",
+            sf_client_secret="test",
+            quota_percent_total=None,
+            quota_percent_per_run=None,
+            is_sandbox=None,
+            select_fields_by_default= False,
+            default_start_date='2019-02-04T12:15:00Z',
+            api_type="BULK")
```

Comment:
```suggestion
        sf = Salesforce(default_start_date='2019-02-04T12:15:00Z', api_type="BULK")
```

The tests still run without all of this extra code, so we shouldn't clutter the file with it to help increase the readability of the test

Reply:
Updated the code to set minimal arguments.